dune_client/api/base.py

```diff
@@ -0,0 +1,115 @@
+""""
+Basic Dune Client Class responsible for refreshing Dune Queries
+Framework built on Dune's API Documentation
+https://duneanalytics.notion.site/API-Documentation-1b93d16e0fa941398e15047f643e003a
+"""
+from __future__ import annotations
+
+import logging.config
+import os
+from json import JSONDecodeError
+from typing import Dict, Optional, Any
+
+import requests
+from requests import Response
+
+
+# pylint: disable=too-few-public-methods
+class BaseDuneClient:
+    """
+    A Base Client for Dune which sets up default values
+    and provides some convenient functions to use in other clients
+    """
+
+    BASE_URL = "https://api.dune.com"
+    DEFAULT_TIMEOUT = 10
+
+    def __init__(
+        self, api_key: str, client_version: str = "v1", performance: str = "medium"
+    ):
+        self.token = api_key
+        self.client_version = client_version
+        self.performance = performance
+        self.logger = logging.getLogger(__name__)
+        logging.basicConfig(format="%(asctime)s %(levelname)s %(name)s %(message)s")
+
+    @classmethod
+    def from_env(cls) -> BaseDuneClient:
+        """
+        Constructor allowing user to instantiate a client from environment variable
+        without having to import dotenv or os manually
+        We use `DUNE_API_KEY` as the environment variable that holds the API key.
+        """
+        return cls(os.environ["DUNE_API_KEY"])
+
+    @property
+    def api_version(self) -> str:
```

Comment:
I can get rid of it, but I will have to check an see what it was being used for in the first place.

Reply:
Looking closer here, it seems like something that should be refactored but it is also tethered to the `alpha/v1` route that was temporarily made available for testing CRUD operations. It is currently a "fixed" attribute of the `BaseClient`, but would have to be refactored to be route specific. 

I agree that this refactoring should be done, but it might be worth waiting until there actually is a v2 version.